app/shop.py

```diff
@@ -0,0 +1,8 @@
+class Shop:
+    def __init__(self, shop: dict) -> None:
+        self.name = shop["name"]
+        self.location = shop["location"]
+        self.products = shop["products"]
+        self.price_milk = self.products["milk"]
+        self.price_bread = self.products["bread"]
+        self.price_butter = self.products["butter"]

```

Comment:
Consider adding error handling for missing keys in the `shop` dictionary. Using `dict.get()` with default values can prevent `KeyError` exceptions if any expected key is missing.